### PR TITLE
Feature/TD-1356,TD-1238-generic-dialog-component

### PR DESCRIPTION
**Pull request by m4manjesh**
This PR contributes to 
https://sce.myjetbrains.com/youtrack/issue/TD-1238/Model-overview-detail-view-add-vehicle
https://sce.myjetbrains.com/youtrack/issue/TD-1356/Model-overview-detail-view-edit-model

- Based on the team discussion on Friday created a generic dialog component  with "Save"  and "Cancel" actions ( text can be passed as props)
- We now have a prop called "content" to pass any valid react element. 
- Close button icon added to dialog after discussion with @Sowbhagya-ipg , by default it will be shown. Can manage with "showCloseIcon" prop

This will be integrated into the MM edit modal and Add Vehicle.

Figma reference
screenshot-www figma com-2023 04 04-15_03_33

implemented UI
screenshot-localhost_6006-2023 04 04-15_12_27

### CODE REVIEW

File: src/ActionDialog/ActionDialog.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          disabled={saveEnabled}
+        >
+          {saveText}
+        </Button>
+      </DialogActions>
+    </ActionDialogWrapper>
+  );
+}
+
+ActionDialog.propTypes = {
```
Comment (jegasriskantha) [inline src/ActionDialog/ActionDialog.js:82]: The proptypes here are not complete. 

Missing title, cancelText, saveText and open

File: src/ActionDialog/ActionDialog.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+import CloseIcon from "@mui/icons-material/Close";
+import PropTypes from "prop-types";
+import { styled } from "@mui/material/styles";
+
+/**
+ * Horizontal alignment button group component to toggle between left/center/right/justify alignment
+ */
+export default function ActionDialog({
+  onCancelClick = () => {},
```
Comment (jegasriskantha) [inline src/ActionDialog/ActionDialog.js:21]: It would be nicer if these props were in alphabetical order.

**Review by jegasriskantha — CHANGES_REQUESTED** [review]
A couple of minor points but otherwise looks good. 

- [ ] Update props so that they are in alphabetical order 
- [ ] Update saveEnabled logic
- [ ] Add missing propTypes

File: src/ActionDialog/ActionDialog.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      </DialogTitle>
+      <DialogContent dividers>
+        <Stack spacing={3}>{content}</Stack>
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={onCancelClick}>{cancelText}</Button>
+        <Button
+          variant="contained"
+          onClick={onSaveClick}
+          disabled={saveEnabled}
```
[the remark this answers is not in the document]
Comment (m4manjesh) [reply src/ActionDialog/ActionDialog.js]: renamed to saveDisabled

File: src/ActionDialog/ActionDialog.js (hunk 2 above)
Comment (m4manjesh) [reply src/ActionDialog/ActionDialog.js:21]: Sorted

File: src/ActionDialog/ActionDialog.js (hunk 1 above)
Comment (m4manjesh) [reply src/ActionDialog/ActionDialog.js:83]: Added

**Comment by jegasriskantha** [comment]
LGTM. I'll leave to you to release and merge.

**Review by jegasriskantha — APPROVED** [review]

**Pull request merged**